src/modules/m_cgiirc.cpp

```diff
@@ -317,6 +319,7 @@ class ModuleCgiIRC : public Module
 			if(InspIRCd::Match(user->host, iter->hostmask, ascii_case_insensitive_map) || InspIRCd::MatchCIDR(user->GetIPString(), iter->hostmask, ascii_case_insensitive_map))
 			{
 				// Deal with it...
+				user->exempt = (ServerInstance->XLines->MatchesLine("E",user) != NULL);
```

Comment:
This is in the wrong place, at the time this runs the user IP etc. hasn't been changed yet